tests/test_modules.py

```diff
@@ -36,6 +36,37 @@ class Foo(proto.Message):
         del sys.modules[__name__].__protobuf__
 
 
+def test_module_package_cross_api():
+    sys.modules[__name__].__protobuf__ = proto.module(package="spam.eggs.v1")
+    try:
+
+        class Baz(proto.Message):
+            foo = proto.RepeatedField(proto.INT64, number=1)
+
+        marshal = proto.Marshal(name="spam.eggs.v1")
+
+        assert Baz.meta.package == "spam.eggs.v1"
+        assert Baz.pb() in marshal._rules
+
+        sys.modules[__name__].__protobuf__ = proto.module(package="ham.pancakes.v1")
+
+        class AnotherMessage(proto.Message):
+            qux = proto.Field(proto.MESSAGE, number=1, message=Baz)
+
+        marshal = proto.Marshal(name="ham.pancakes.v1")
+
+        assert AnotherMessage.meta.package == "ham.pancakes.v1"
+        assert AnotherMessage.pb() in marshal._rules
+        # Confirm that Baz.pb() is no longer present in marshal._rules
+        assert Baz.pb() not in marshal._rules
+
+        # Test using multiple packages together
+        msg = AnotherMessage(qux=Baz())
+        assert type(msg) == AnotherMessage
```

Comment:
So the error would show up instantiation time in line 64, right? Could you add a brief comment to that effect, with either the error we would get or at least a link to https://github.com/googleapis/python-gke-hub/issues/151?

(without the context it's a bit confusing)

Reply:
Correct, the stack trace is in the bug that is linked: https://github.com/googleapis/proto-plus-python/issues/349. I'll add a link to the bug above line 64.

Reply:
Done.